build.gradle

```diff
@@ -9,311 +9,401 @@ buildscript {
+            '**/brs/at/AtControllerTest.class'
+}
+
+application {
+    mainClass.set('signum.Launcher')
 }
 
 sourceSets {
-  main {
-    java {
-      srcDir "src"
-    }
-    resources {
-      srcDir "resources"
+    main {
+        java.srcDir "src"
+        resources.srcDir "resources"
     }
-  }
-  if (!project.hasProperty('skipTests')) {
-    test {
-      java {
-        srcDirs = ["test/java"]
-      }
+    if (!project.hasProperty('skipTests')) {
+        test {
+            java.srcDirs = ["test/java"]
+        }
     }
-  }
 }
 
-// Define the main class for the application
-mainClassName = 'signum.Launcher'
-description = 'Signum Node.'
-
 jar {
-  // Needed by the logging library
-  manifest {
-    attributes 'Multi-Release': 'true'
-  }
+    manifest {
+        attributes 'Multi-Release': 'true'
+    }
 }
 
 shadowJar {
-  mergeServiceFiles()
+    mergeServiceFiles()
 }
 
-def details = versionDetails()
-def baseNodeFileName = "signum-node"
-createExe {
-  outfile = "${baseNodeFileName}.exe"
-  icon = "${projectDir}/resources/images/signum_overlay_logo.ico"
-  copyright = 'signum.network'
-  companyName = 'https://signum.network'
-  textVersion = details.lastTag
-  maxHeapSize = 2048
-
-  bundledJrePath = 'jre/'
-
-  dontWrapJar = true
-  mainClassName = project.mainClassName
-  copyConfigurable = project.tasks.shadowJar.outputs.files
-  jar = "${baseNodeFileName}.jar"
+tasks.register('buildOpenApi', NpmTask) {
+    dependsOn npmInstall
+    args = ['run', 'dist']
 }
 
 node {
-  download = true
-  version = '16.14.0'
-  nodeProjectDir = file("${project.projectDir}/openapi")
-  npmInstallCommand = "ci"
+    download = true
+    version = '20.19.1'
+    nodeProjectDir = file("${project.projectDir}/openapi")
+    npmInstallCommand = "ci"
 }
 
-tasks.register("buildOpenApi", NpmTask) {
-  dependsOn npmInstall
-  args = ['run', 'dist']
+tasks.register('dist', Zip) {
+    from(project.tasks.shadowJar.outputs.files) {
+        include "${baseNodeFileName}-all.jar"
+        rename { "${baseNodeFileName}.jar" }
+    }
+    into('conf') { from "conf/node-default.properties" }
+    into('conf') { from "conf/logging-default.properties" }
+    into('html') { from "html" }
+    from "LICENSE.txt"
+    from "README.md"
+    archiveFileName = "${baseNodeFileName}.zip"
 }
 
-tasks.register('dist', Zip) {
-  dependsOn buildOpenApi
-  from(project.tasks.shadowJar.outputs.files) {
-    include "${baseNodeFileName}-all.jar"
-    rename { "${baseNodeFileName}.jar" }
-  }
-  into('conf') { from "conf/node-default.properties" }
-  into('conf') { from "conf/logging-default.properties" }
-  into('html') { from "html" }
-  from "LICENSE.txt"
-  from "README.md"
-  archiveName "${baseNodeFileName}.zip"
+def appName = "SignumNode"
+
+tasks.register('jpackageWin', Exec) {
+    group = 'distribution'
+    description = 'Builds a Windows EXE using jpackage'
+    dependsOn shadowJar
+
+    def rawVersion = details.lastTag ?: "1.0.0"
```

Comment:
👀 WTF? Why such complicated logic? Isn't there something more straightforward?

Reply:
? it is more or less the same :-)

Reply:
ah, you mean the version stuff?  Yes seems jpackage has a problem with some numbers to high or not in the "right" pattern